### ISSUE TITLE
Useless ternary as a separate sniff

### ISSUE THREAD

**Issue by VasekPurchart** [slevomat/coding-standard#556]
Right now `UselessConditionWithReturn` detects useless ternary operators, but only if they are part of an return statement:

```php
<?php
// detected
return $foo && $bar ? true : false;

// not detected
$test = $foo && $bar ? true : false;
```

So this logic could be separated to a standalone sniff to be generally applicable.

Also there is now a bug where if the condition is within return statement, but in different scope, the detection actually works:

```php
<?php
return test(function () {
        // detected
	$test = $foo && $bar ? true : false;

	return $test;
});
```

So it already works pretty well for these cases :)

**Comment by TomasVotruba** [comment]
@kukulich Why did you delete the comment?

**Comment by kukulich** [comment]
Bug fixed in https://github.com/slevomat/coding-standard/commit/6588f42fc9ee257a52811adf16114cb771dd7336
